sdk/core/logger/src/logger.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import debug from "debug";
+
+const registeredLoggers = new Set<AzureDebugger>();
+let azureLogLevel: AzureLogLevel | undefined =
+  (typeof process !== "undefined" &&
+    process.env &&
+    (process.env.AZURE_LOG_LEVEL as AzureLogLevel)) ||
+  undefined;
+export const AzureLogger = debug("azure") as AzureDebugger;
+AzureLogger.log = (...args) => {
+  debug.log(...args);
+};
+
+export type AzureLogLevel = "verbose" | "info" | "warning" | "error";
+
+type AzureDebugger = debug.Debugger & { level: number };
+
+if (typeof process !== "undefined" && process.env && process.env.AZURE_LOG_LEVEL) {
```

Comment:
line 7 already does something similar, can we not consolidate?

Reply:
Done